modules/social_features/social_group/src/Plugin/Field/FieldWidget/SocialGroupSelectorWidget.php

```diff
@@ -111,79 +112,103 @@ public static function create(ContainerInterface $container, array $configuratio
+        $permission = 'administer nodes';
+      }
+      else {
+        $definition = $this->entityTypeManager->getDefinition($type);
+
+        if ($definition !== NULL) {
+          $permission = $definition->getAdminPermission();
+        }
+      }
+
+      if (
+        empty($permission) ||
+        !is_string($permission) ||
+        !$account->hasPermission($permission)
+      ) {
+        $account = $entity->getOwner();
+      }
     }
 
     // Limit the settable options for the current user account.
-    $options = $this->fieldDefinition
+    $options_provider = $this->fieldDefinition
       ->getFieldStorageDefinition()
-      ->getOptionsProvider($this->column, $entity)
-      ->getSettableOptions($account);
-
-    // Check for each group type if the content type is installed.
-    foreach ($options as $key => $optgroup) {
-      // Groups are in the array below.
-      if (is_array($optgroup)) {
-        // Loop through the groups.
-        foreach ($optgroup as $gid => $title) {
+      ->getOptionsProvider($this->column, $entity);
+
+    if ($options_provider !== NULL) {
+      $options = $options_provider->getSettableOptions($account);
+
+      $storage = $this->entityTypeManager->getStorage('group');
+
+      // Check for each group type if the content type is installed.
+      foreach ($options as $key => $optgroup) {
+        // Groups are in the array below.
+        if (is_array($optgroup)) {
+          $group = $storage->load(array_keys($optgroup)[0]);
+
           // If the group exists.
-          if ($group = Group::load($gid)) {
-            // Load all installed plugins for this group type.
-            $plugin_ids = $this->pluginManager->getInstalledIds($group->getGroupType());
+          if ($group instanceof GroupInterface) {
+            $supported = $this->pluginManager
+              ->getInstalled($group->getGroupType())
+              ->has('group_' . $type . ':' . $entity->bundle());
+
             // If the bundle is not installed,
             // then unset the entire optiongroup (=group type).
-            if (!in_array('group_node:' . $entity_type, $plugin_ids)) {
+            if (!$supported) {
               unset($options[$key]);
             }
           }
-          // We need to check only one of each group type,
-          // so break out the second each.
-          break;
         }
       }
-    }
 
-    // Remove groups the user does not have create access to.
-    if (!$account->hasPermission('manage all groups')) {
-      $options = $this->removeGroupsWithoutCreateAccess($options, $account, $entity);
+      // Remove groups the user does not have create access to.
+      if (!$groups_admin) {
+        $options = $this->removeGroupsWithoutCreateAccess(
+          $options,
+          $account,
+          $entity,
+        );
+      }
+    }
+    else {
+      $options = [];
     }
 
     // Add an empty option if the widget needs one.
     if ($empty_label = $this->getEmptyLabel()) {
```

Comment:
What happens if `$options` is [] here?

Reply:
When you will try to create an entity inside a group and this array isn't filled in by the "options_list" alter hook below then you should see only the "- None -" item in the "Group" field and when you press "Save" then the entity will be created outside of the group. An existing entity will be moved outside the group when you will edit the first one. I don't think that that array will be empty at any time. I've implemented it just to support a possible empty value returned by the "getOptionsProvider" method and avoid the new ignoring rule for PHPStan.